Mathlib/AlgebraicTopology/SimplicialSet/HornColimits.lean

```diff
@@ -123,4 +123,171 @@ lemma isPushout :
 
 end horn₂₂
 
+namespace horn
+
+variable {n : ℕ} (i : Fin (n + 1))
+
+/-- The multicoequalizer diagram which expresses `Λ[n, i]` as a gluing
+of all `1`-codimensional faces of the standard simplex but one
+along suitable `2`-codimensional faces. -/
+lemma multicoequalizerDiagram :
+    Subcomplex.MulticoequalizerDiagram Λ[n, i]
+      (ι := ({i}ᶜ : Set (Fin (n +1)))) (fun j ↦ stdSimplex.face {j.1}ᶜ)
+      (fun j k ↦ stdSimplex.face {j.1, k.1}ᶜ) where
+  iSup_eq := by rw [horn_eq_iSup]
+  eq_inf j k := by
+    rw [stdSimplex.face_inter_face]
+    congr
+    aesop
+
+/-- The horn is a multicoequalizer of all `1`-codimensional faces of the
+standard simplex but one along suitable `2`-codimensional faces. -/
+noncomputable def isColimit :
+    IsColimit ((multicoequalizerDiagram i).multicofork.toLinearOrder.map
+      Subcomplex.toSSetFunctor) :=
+  (multicoequalizerDiagram i).isColimit'
+
+end horn
+
+namespace horn₃₁
+
+/-- The inclusion `Δ[2] ⟶ Λ[3, 1]` which avoids `0`. -/
+abbrev ι₀ : Δ[2] ⟶ Λ[3, 1] := horn.ι.{u} 1 0 (by simp)
+
+/-- The inclusion `Δ[2] ⟶ Λ[3, 1]` which avoids `2`. -/
+abbrev ι₂ : Δ[2] ⟶ Λ[3, 1] := horn.ι.{u} 1 2 (by simp)
+
+/-- The inclusion `Δ[2] ⟶ Λ[3, 1]` which avoids `3`. -/
+abbrev ι₃ : Δ[2] ⟶ Λ[3, 1] := horn.ι.{u} 1 3 (by simp)
+
+variable {X : SSet.{u}} (f₀ f₂ f₃ : Δ[2] ⟶ X)
+  (h₁₂ : stdSimplex.δ 2 ≫ f₀ = stdSimplex.δ 0 ≫ f₃)
+  (h₁₃ : stdSimplex.δ 1 ≫ f₀ = stdSimplex.δ 0 ≫ f₂)
+  (h₂₃ : stdSimplex.δ 2 ≫ f₂ = stdSimplex.δ 2 ≫ f₃)
+
+/-- Auxiliary definition for `desc`. -/
+@[simps!]
+def desc.multicofork :
+    Multicofork ((horn.multicoequalizerDiagram (1 : Fin 4)).multispanIndex.toLinearOrder.map
+      Subcomplex.toSSetFunctor) :=
+  Multicofork.ofπ _ X (fun ⟨(i : Fin 4), hi⟩ ↦ match i with
+    | 0 => (stdSimplex.faceSingletonComplIso 0).inv ≫ f₀
+    | 1 => by simp at hi
```

Comment:
```suggestion
    | 1 => False.elim <| by simp at hi
```
Purely for readability: this way it’s clear at an immediate glance at the code that this is an impossible case, rather than triggering a "why is there a `by simp …` supposedly producing data?" initial reaction